docs/upgrade.html

```diff
@@ -21,6 +21,10 @@
 0.10.0.0 has <a href="#upgrade_10_breaking">potential breaking changes</a> (please review before upgrading) and
 there may be a <a href="#upgrade_10_performance_impact">performance impact during the upgrade</a>. Because new protocols
 are introduced, it is important to upgrade your Kafka clusters before upgrading your clients.
+<p/>
+<b>Notes to client with version 0.9.0.0: </b>Due to a bug introduced in 0.9.0.0,
+clients with ZooKeeper dependency may not be able to work with 0.10.0.0 broker. Therefore, 0.9.0.0 clients should
```

Comment:
I agree, we don't want to be limited to just adding fields, but when we are just adding fields a version bump may not be required. This patch assumes anything over version 1 will be compatible in the else statement, so even though our goal is being able to remove fields, this patch does not do that. 

I don't think adding another version makes sense, thats just adding another field to (old) version 2. We can just add the rack field and achieve the same goal.